File: ios/Features/Home/HomeViewModel.swift
import Foundation
import Combine

class HomeViewModel: ObservableObject {
    @Published var stats = DashboardStats(liveChannels: 0, movies: 0, series: 0)
    @Published var recentChannels: [Channel] = []
    @Published var playlists: [Playlist] = []
    @Published var isLoading = false
    @Published var error: String?
    
    private let repository = AppRepository()
    private var cancellables = Set<AnyCancellable>()
    
    // Cloudflare Worker URL - update sau khi deploy
    private let defaultPlaylistUrl = "https://playertv-app.YOUR_SUBDOMAIN.workers.dev/playlist.m3u"
    
    init() {
        loadDefaultPlaylistIfNeeded()
        loadData()
    }
    
    private func loadDefaultPlaylistIfNeeded() {
        // Check if first launch
        if repository.playlists.isEmpty {
            Task {
                do {
                    _ = try await repository.importM3uFromUrl(defaultPlaylistUrl, name: "PlayerTV (Default)")
                } catch {
                    // Fallback to built-in defaults
                    importBuiltInDefaults()
                }
            }
        }
    }
    
    private func importBuiltInDefaults() {
        // Removed hardcoded channels
    }
    
    private func loadData() {
        repository.$playlists
            .receive(on: DispatchQueue.main)
            .sink { [weak self] playlists in
                self?.playlists = playlists
                let totalChannels = playlists.reduce(0) { $0 + $1.channelCount }
                self?.stats = DashboardStats(
                    liveChannels: totalChannels,
                    movies: 0,
                    series: 0
                )
            }
            .store(in: &cancellables)
    }
}

File: android/app/src/main/kotlin/com/player/tv/ui/navigation/HomeViewModel.kt
package com.player.tv.ui.navigation

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.player.tv.data.local.RecentEntity
import com.player.tv.data.remote.M3UParser
import com.player.tv.data.repository.AppRepository
import com.player.tv.domain.model.DashboardStats
import com.player.tv.domain.model.Playlist
import com.player.tv.util.Constants
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.*
import kotlinx.coroutines.launch
import javax.inject.Inject

data class HomeUiState(
    val stats: DashboardStats = DashboardStats(),
    val recentChannels: List<RecentEntity> = emptyList(),
    val playlists: List<Playlist> = emptyList(),
    val isLoading: Boolean = false,
    val error: String? = null
)

@HiltViewModel
class HomeViewModel @Inject constructor(
    private val repository: AppRepository,
    private val m3uParser: M3UParser
) : ViewModel() {

    private val _uiState = MutableStateFlow(HomeUiState())
    val uiState: StateFlow<HomeUiState> = _uiState.asStateFlow()

    init {
        loadDefaultPlaylistIfNeeded()
        loadData()
    }

    private fun loadDefaultPlaylistIfNeeded() {
        viewModelScope.launch {
            repository.getAllPlaylists().first().let { playlists ->
                if (playlists.isEmpty()) {
                    // Try secure API first
                    try {
                        val channels = m3uParser.fetchFromSecureApi()
                        if (channels.isNotEmpty()) {
                            val m3uContent = buildString {
                                appendLine("#EXTM3U")
                                channels.forEach { ch ->
                                    appendLine("#EXTINF:-1 tvg-name=\"${ch.name}\" group-title=\"${ch.groupTitle ?: ""}\",${ch.name}")
                                    appendLine(ch.url)
                                }
                            }
                            repository.importM3uFromFile(m3uContent, "PlayerTV")
                        } else {
                            importBuiltInDefaults()
                        }
                    } catch (e: Exception) {
                        importBuiltInDefaults()
                    }
                }
            }
        }
    }

    private fun importBuiltInDefaults() {
        // Removed
    }

    private fun loadData() {
        viewModelScope.launch {
            repository.getAllPlaylists().collect { playlists ->
                val totalChannels = playlists.sumOf { it.channelCount }
                _uiState.update { state ->
                    state.copy(
                        playlists = playlists,
                        stats = DashboardStats(
                            liveChannels = totalChannels,
                            movies = 0,
                            series = 0
                        )
                    )
                }
            }
        }

        viewModelScope.launch {
            repository.getRecentChannels().collect { recent ->
                _uiState.update { it.copy(recentChannels = recent) }
            }
        }
    }
}
